light_controller/ir_toy/ir_toy.py

```diff
@@ -0,0 +1,351 @@
+# -*- coding: utf-8 -*-
+import logging
+import time
+import struct
+
+import serial
+
+from .ir_toy_settings import IR_ToySettings
+
+__author__ = 'akm'
+
+
+class CommandEnum:
+    reset = 0x00
+    SUMP_Run = 0x01
+    SUMP_Id = 0x02
+    transmit = 0x03
+    getFrequencyReport = 0x04
+    setSampleTimer = 0x05
+    setFrequencyModulationTimer = 0x06
+    setLED_MuteOn = 0x10
+    setLED_MuteOff = 0x11
+    setLED_On = 0x12
+    setLED_Off = 0x13
+    getSettingsReport = 0x23
+    enableTX_ByteReport = 0x24
+    enableTX_NotifyOnComplete = 0x25
+    enableTX_Handshake = 0x26
+    IO_Write = 0x30
+    IO_Direction = 0x31
+    IO_Read = 0x32
+    IO_Direction = 0x31
+    setupUART = 0x40
+    closeUART = 0x41
+    writeUART = 0x42
+
+
+class IR_Toy:
```

Comment:
We may want to handle the failure cases and not proceed to next operation if a failure occurs and silently ignore it - but rather fail early and signal to caller.